### PR TITLE
open up react and react-native peerDeps

### PR DESCRIPTION
**Pull request by alexburkowskypolysign**
actually configure peerDeps as the readme states per https://github.com/pixielabs/cavy#-requirements

Issue: https://github.com/pixielabs/cavy/issues/226

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,9 +10,11 @@ jobs:
       - run:
           name: Install React Native dependencies
           command: |
+            git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core fetch --unshallow
+            git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask fetch --unshallow
```
Comment (alexburkowskypolysign) [inline .circleci/config.yml:14]: brew update breaks without these

**Comment by jalada** [comment]
Hi @alexburkowskypolysign, I wonder if this is more a case of an incorrect README; there's no guarantees that Cavy will work with all versions of RN & React greater than the versions defined in the README (or, with this PR, in the peer dependencies). In the past we've had to do a lot of work to bring Cavy up to date with new versions of RN & React (something that we are due to do, as we haven't done it for a while).

What do you think?

**Comment by alexburkowskypolysign** [comment]
@jalada We have tested this with the latest react and react-native versions in our repo over at Polysign and all works well. I cant guarantee this will work indefinitely for all future versions, but I dont think that should be the expectation. Many packages open up their peer dependencies to avoid conflicts with npm7 new peer dependency management

**Comment by alexburkowskypolysign** [comment]
@jalada thoughts on limiting to an upper range?

**Comment by jalada** [comment]
Hey @alexburkowskypolysign yeah I like the idea of limiting to an upper range.

**Comment by alexburkowskypolysign** [comment]
@jalada all set. Please lmk if there is anything else needed for merge

**Pull request merged**